cirq/circuits/circuit.py

```diff
@@ -211,15 +211,51 @@ def __getitem__(self, key: slice) -> 'Circuit':
     def __getitem__(self, key: int) -> 'cirq.Moment':
         pass
 
+    @overload
+    def __getitem__(self, key: Tuple[int, 'cirq.Qid']) -> 'cirq.Operation':
+        pass
+
+    @overload
+    def __getitem__(self,
+                    key: Tuple[int, Iterable['cirq.Qid']]) -> 'cirq.Moment':
+        pass
+
+    @overload
+    def __getitem__(self, key: Tuple[slice, 'cirq.Qid']) -> 'cirq.Circuit':
+        pass
+
+    @overload
+    def __getitem__(self,
+                    key: Tuple[slice, Iterable['cirq.Qid']]) -> 'cirq.Circuit':
+        pass
+
     def __getitem__(self, key):
         if isinstance(key, slice):
             sliced_circuit = Circuit(device=self.device)
             sliced_circuit._moments = self._moments[key]
             return sliced_circuit
         if isinstance(key, int):
             return self._moments[key]
-
-        raise TypeError('__getitem__ called with key not of type slice or int.')
+        if isinstance(key, Tuple):
+            assert len(key) == 2
+            moment_idx, qubit_idx = key
+            # moment_idx - int or slice; qubit_idx - Qid or Iterable[Qid].
+            if isinstance(moment_idx, int):
+                return self._moments[moment_idx][qubit_idx]
+            if isinstance(moment_idx, slice):
+                if isinstance(qubit_idx, cirq.Qid):
+                    qubit_idx = [qubit_idx]
+                moments = [
+                    moment[qubit_idx] for moment in self._moments[moment_idx]
+                ]
+                moments = [
+                    moment for moment in moments if len(moment.operations) > 0
+                ]
```

Comment:
Remove this.  It should keep empty moments.  A few of the tests will need to be changed.

Reply:
Got it.